File: Python/MinStack.py
class MinStack:

    def __init__(self) -> None:
        self.minStack = []
        self.minArray = []
        self.minElement = 0

    def push(self,val) -> None:
        if len(self.minStack) == 0:
            self.minElement = val
            self.minArray.append(val)
        else:
            if val <= self.minElement:
                self.minElement = val
                self.minArray.append(val)

        self.minStack.append(val)

    def pop(self) -> None:
        topElement = self.minStack[-1]
        if topElement == self.minElement:
            self.minArray.pop()
            if len(self.minArray) !=0:
                self.minElement = self.minArray[-1]
        
        self.minStack.pop()


    def top(self) -> int:
        return self.minStack[-1]


    def getMin(self) -> int:
        return self.min

File: Javascript/MinStack.js
var MinStack = function(){
    this.minStack = []
    this.minArray = []
    this.minElement = 0
}


MinStack.prototype.push = function(val){
    if(this.minStack.length === 0){
        this.minElement = val
        this.minArray.push(val)
    }
    else{
        if(val <= this.minElement){
            this.minElement = val
            this.minArray.push(val)
        }
    }

    this.minStack.push(val)
}

MinStack.prototype.pop = function(){
    let topElement = this.minStack[this.minStack.length - 1]
    if(topElement === this.minElement){
        this.minArray.pop()
        if(this.minArray.length !== 0){
            this.minElement = this.minArray[this.minArray.length - 1]
        }
        
    }
    this.minStack.pop()
}

MinStack.prototype.top = function(){
    return this.minStack[this.minStack.length - 1]
}

MinStack.prototype.getMin = function(){
    return this.minElement
}